pkg/kn/flags/podspec_helper.go

```diff
@@ -255,6 +258,11 @@ func UpdateImagePullSecrets(spec *corev1.PodSpec, pullsecrets string) {
 	}
 }
 
+// UpdateContainers updates the containers array with additional ones provided from file or os.Stdin
+func UpdateContainers(spec *corev1.PodSpec, containers []corev1.Container) {
+	spec.Containers = append(spec.Containers, containers...)
+}
+
```

Comment:
@rhuss the current mode is append additional containers. However, that might not work very well for update operation. I wonder if replacing whatever `spec.Containers[1:]` is a good solution.
Additionally we could require container's `name` parameter and match those if Containers array is already populated.

Reply:
That's a good question. Updating a list is always tricky, but I like the idea with correlating via name. "update" is always an incremental operation, you don't have to specify the full state (that is different to "apply"). But if we would like to update based on position in the list, you would need to be able either to specify the index of the container within this list or provide always all containers to update. I think it is fair to require the container name (and I would even enforce this on creation). If we update a list with containers where not all containers have a name (e.g. on existing services that thas not been created with kn), then we would just ignore those containers on an update.

wdyt ? is this a fair solution ? @dprotaso as kind of stakeholder for this feature ;-) would it be ok to require a name for a container that is managed by kn in a multi-container setup ?